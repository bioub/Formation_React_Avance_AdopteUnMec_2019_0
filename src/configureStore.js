import { createStore, combineReducers, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import { todosReducer } from './todos/reducers';
import { usersReducer } from './users/reducers';

const rootReducer = combineReducers({
  todos: todosReducer,
  users: usersReducer,
});

export function configureStore() {
  const store = createStore(
    rootReducer,
    composeWithDevTools(applyMiddleware(thunk)),
  );

  return store;
}
