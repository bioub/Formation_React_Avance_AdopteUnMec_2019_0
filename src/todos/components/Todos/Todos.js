import React from 'react';
import { TodoFormContainer } from '../../containers/TodoFormContainer';
import { TodoListContainer } from '../../containers/TodoListContainer';
import { HideableClock } from '../../../components/Clock/Clock';

export function Todos() {
  return (
    <div className="Todos">
      <TodoFormContainer />
      <TodoListContainer />

      <hr />
      <HideableClock />
    </div>
  );
}
