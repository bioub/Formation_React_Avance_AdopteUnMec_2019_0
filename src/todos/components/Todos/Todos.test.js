import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { Todos } from './Todos';
import { configureStore } from '../../../configureStore';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  const store = configureStore();
  ReactDOM.render(
    <Provider store={store}>
      <Todos />
    </Provider>,
    div,
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('shallow renders without crashing', () => {
  const renderer = new ShallowRenderer();

  renderer.render(<Todos />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});

it('shallow renders without crashing (enzyme)', () => {
  shallow(<Todos />);
});
