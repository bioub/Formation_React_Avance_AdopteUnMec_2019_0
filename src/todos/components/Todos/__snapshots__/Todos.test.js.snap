// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders without crashing 1`] = `
<div
  className="Todos"
>
  <Memo(ConnectFunction) />
  <Memo(ConnectFunction) />
  <hr />
  <Clock />
</div>
`;
